### PR TITLE
Python 3.8 compatibility

### PR DESCRIPTION
**Pull request by mehrdadn**
## Related issue number

#7587

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -162,7 +162,7 @@ matrix:
 
         - ./ci/suppress_output ./ci/travis/install-dependencies.sh
         # This command should be kept in sync with ray/python/README-building-wheels.md.
-        - ./ci/suppress_output ./python/build-wheel-macos.sh
+        - ./python/build-wheel-macos.sh
```
Comment (simon-mo) [inline .travis.yml]: Can we revert this line?

File: ci/travis/test-wheels.sh [hunk 2]
```diff
@@ -45,15 +46,18 @@ if [[ "$platform" == "linux" ]]; then
 
   # Run a simple test script to make sure that the wheel works.
   INSTALLED_RAY_DIRECTORY=$(dirname "$($PYTHON_EXE -u -c "import ray; print(ray.__file__)" | tail -n1)")
-  $PYTHON_EXE "$TEST_SCRIPT"
+
```
Comment (simon-mo) [inline ci/travis/test-wheels.sh]: @mehrdadn I did something super lazy here. It turns out we are not testing all Python version in linux build. Can you add that here? Thanks!

**Comment by simon-mo** [comment]
I think there's a bug in the bash for loop somehow? the wheel variable end up being all four wheels
```
PYTHON_WHEEL='/home/travis/build/ray-project/ray/ci/travis/../../.whl/ray-0.9.0.dev0-cp37-cp37m-manylinux1_x86_64.whl
/home/travis/build/ray-project/ray/ci/travis/../../.whl/ray-0.9.0.dev0-cp36-cp36m-manylinux1_x86_64.whl
/home/travis/build/ray-project/ray/ci/travis/../../.whl/ray-0.9.0.dev0-cp38-cp38-manylinux1_x86_64.whl
/home/travis/build/ray-project/ray/ci/travis/../../.whl/ray-0.9.0.dev0-cp35-cp35m-manylinux1_x86_64.whl'
```

**Comment by mehrdadn** [comment]
Pushed (hopefully) a fix, thanks!

File: ci/travis/test-wheels.sh [hunk 3]
```diff
@@ -21,41 +21,53 @@ else
...
 if [[ "$platform" == "linux" ]]; then
-  # Now test Python 3.6.
-
   # Install miniconda.
-  wget --quiet https://repo.continuum.io/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh -O miniconda3.sh
+  PY_MMS=("3.6"
+          "3.7"
+          "3.8.2")
+  wget --quiet "https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh" -O miniconda3.sh
   bash miniconda3.sh -b -p "$HOME/miniconda3"
```
Comment (simon-mo) [inline ci/travis/test-wheels.sh]:
```suggestion
export PATH=$HOME/miniconda3/bin:$PATH
```

Comment (mehrdadn) [reply ci/travis/test-wheels.sh]: Done!

**Review by simon-mo — APPROVED** [review]
LGTM!!

**Pull request merged**